impl/src/main/java/org/jboss/weld/util/BeanMethods.java

```diff
@@ -325,7 +324,13 @@ protected void duplicateMethod(EnhancedAnnotatedMethod<?, ? super T> method) {
      * @see https://issues.jboss.org/browse/WELD-2019
      */
     public static <T> Collection<EnhancedAnnotatedMethod<?, ? super T>> filterMethods(final Collection<EnhancedAnnotatedMethod<?, ? super T>> methods) {
-        return Collections2.filter(methods, METHOD_FILTER_PREDICATE);
+        List<EnhancedAnnotatedMethod<?, ? super T>> filteredMethods = new ArrayList<>();
+        for (EnhancedAnnotatedMethod<?, ? super T> method : methods) {
+            if (!method.getJavaMember().isBridge() && !method.getJavaMember().isSynthetic()) {
+                filteredMethods.add(method);
+            }
+        }
+        return ImmutableList.copyOf(filteredMethods);
```

Comment:
I wonder whether `java.util.Collections.unmodifiableList(List<? extends T>)` wouldn't be more appropriate (and more performant) here. It seems the returned collection is not stored anywhere.